templates/static/js/scripts.js

```diff
@@ -87,6 +87,12 @@
 
 // Add table of contents to side navigation on documentation pages
 (function () {
+  // Generate id from H2s content when it does not exist
+  document.querySelectorAll('main h2:not([id])').forEach(function (heading) {
+    var id = heading.textContent.toLowerCase().replaceAll(/\s+/g, '-');
```

Comment:
This may fail with some special characters in headers.

Maybe worth adding some more rules in here. That's what I found in one markdown implementation:

https://github.com/markedjs/marked/blob/30e90e5175700890e6feb1836c57b9404c854466/src/Slugger.js#L19